bundles/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/services/DeviceManager.java

```diff
@@ -168,11 +185,14 @@ private void searchDevice() {
                                 logger.debug("{}: got bulbInfo: {}", TAG, bulbInfo);
                                 if (bulbInfo.containsKey("model") && bulbInfo.containsKey("id")) {
                                     DeviceBase device = DeviceFactory.build(bulbInfo);
-                                    if (bulbInfo.containsKey("name")) {
-                                        device.setDeviceName(bulbInfo.get("name"));
-                                    } else {
-                                        device.setDeviceName("");
+
+                                    if (null == device) {
+                                        logger.warn("Found unsupported device.");
+                                        continue;
                                     }
+                                    stopDiscovery();
```

Comment:
Calling stop here means it stops after the first device is found. Is that intended?

Reply:
No, it wasn't intended. I've reverted this line. It was a left over from testing why the discovery didn't work.